source/dagger/gameplay/common/aiming_system.cpp

```diff
@@ -0,0 +1,86 @@
+#define _USE_MATH_DEFINES
+
+#include "aiming_system.h"
+
+#include "core/graphics/sprite.h"
+#include "core/input/inputs.h"
+
+#include <cmath>
+#include <math.h>
+
+void AimingSystem::OnInitialize(Registry& registry_, Entity entity_)
+{
+    InputReceiver& receiver = registry_.get<InputReceiver>(entity_);
+    for (auto command : { "rotate" })      //Input context has to have a key bound for "rotate" command (or you can rename the command to something you like)
+    {
+        receiver.values[command] = 0;
+    }
+}
+
+void AimingSystem::Run()
+{
+    Engine::Registry().view<InputReceiver, Sprite, Crosshair>().each(
+        [](const InputReceiver input_, Sprite& sprite_, Crosshair& crosshair_) //The sprite component with the same entity as this crosshair component is of a sprite that is used for center of rotation (like character for example)
+        {
+        /*
+         Example setup of for "rotate" command in input-context :
+
+                    "command-name": "rotate",
+                       "actions" : [
+                     {
+                       "trigger": "KeyA",
+                       "value" : 0.05
+                     },
+                     {
+                      "trigger": "KeyD",
+                      "value" : -0.05
+                     }
+                    ]
+       */
+            Float32 rotate = input_.values.at("rotate");        //Get the input value of the rotate command - the amount the angle changes if the button is pressed 
```

Comment:
Use `input_.Get("rotate")` instead. Maybe the `values` map should be private in the future...